### PR TITLE
[pickers] Fix `x-date-pickers-pro` theme augmentation

### LINKED ISSUE #4673: [DateRangePicker] How to override `DateRangePickerViewDesktopContainer` styles ?
### Order ID 💳

42474

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I'd like to remove the border from the DateRangePicker, but it appears to be hardcoded 
https://github.com/mui/mui-x/blob/77acb3bddb79e57ae1205cf1e1ecaf8f45674392/packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerViewDesktop.tsx#L53-L57

Is there a way of customising this? The closest I've come is some very fragile CSS, or I imagine I could wrap it with a ThemeProvider and set `theme.palette.divider` to `transparent`?
Thanks

image

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  
  System:
    OS: macOS 11.6
  Binaries:
    Node: 16.10.0 - /usr/local/bin/node
    Yarn: 1.19.0 - /usr/local/bin/yarn
    npm: 7.24.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 100.0.4896.127
    Edge: Not Found
    Firefox: 87.0
    Safari: 15.0
  npmPackages:
    @emotion/react: ^11.8.1 => 11.9.0
    @emotion/styled: ^11.8.1 => 11.8.1
    @mui/material: ^5.4.4 => 5.6.3
    @mui/system: ^5.4.4 => 5.6.3
    @mui/types:  7.1.3
    @mui/x-date-pickers: ^5.0.0-alpha.2 => 5.0.0-alpha.2
    @mui/x-date-pickers-pro: ^5.0.0-alpha.2 => 5.0.0-alpha.2
    @mui/x-license-pro: ^5.10.0 => 5.10.0
    @types/react:  17.0.44
    react: ^17.0.2 => 17.0.2
    react-dom: ^17.0.2 => 17.0.2
    typescript: ^4.1.3 => 4.6.3

```
</details>

### LINKED ISSUE #4942: [DatePicker] `className`s provided via prop are never applied to root element
Not much more to say; the `className` prop is destructured in preparation to apply to the component but never gets used.

https://github.com/mui-org/material-ui/blob/ad2535a8b15f76355f4559dbac02e9cb379269ba/packages/mui-lab/src/internal/pickers/Picker/Picker.tsx#L77

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #4673 
Fixes #4942 

- Fix pro package components theme augmentation (style and prop overrides).
- Apply `static` pickers `className` to root element (for 4942)
- Add some missing base package components to `themeAugmentation`

### CODE REVIEW

File: packages/x-date-pickers-pro/src/DateRangePicker/dateRangePickerToolbarClasses.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export type DateRangePickerToolbarClassKey = keyof DateRangePickerToolbarClasses;
+
+export function getDateRangePickerToolbarUtilityClass(slot: string) {
+  return generateUtilityClass('MuiDateRangePickerToolbar', slot);
+}
+
+export const dateRangePickerToolbarClasses: DateRangePickerToolbarClasses = generateUtilityClasses(
+  'MuiDateRangePickerToolbar',
+  ['root', 'container'],
+);
```
Comment (alexfauquette) [inline packages/x-date-pickers-pro/src/DateRangePicker/dateRangePickerToolbarClasses.ts:19]: I do not see where its used, and it does not seems to be exported for developpers

Comment (LukasTy) [reply packages/x-date-pickers-pro/src/DateRangePicker/dateRangePickerToolbarClasses.ts:19]: You mean that we should export these to be used for nested style overriding? 🤔

Comment (LukasTy) [reply packages/x-date-pickers-pro/src/DateRangePicker/dateRangePickerToolbarClasses.ts:19]: Did I cover what you saw missing with [this commit](https://github.com/mui/mui-x/pull/6096/commits/a2c4a5d94908f3d332075bf4eb92bebee320413b)?

Comment (alexfauquette) [reply packages/x-date-pickers-pro/src/DateRangePicker/dateRangePickerToolbarClasses.ts:19]: Yes, it seems that other classes are exported like that 🤔

I nearly neither use those customization features, so I'm not aware of all the subtilities. I was just curious to know why those classes were added

**Comment by DanailH** [comment]
I remember some discussion about having this one released under `v5`. Is that still the case?

**Comment by LukasTy** [comment]
@DanailH Yes, I'd like for this and following PR for `x-date-pickers` to be in v5 just to make v5 release more open to styling extension. 🤔

**Review by alexfauquette — APPROVED** [review]
Sounds good 🎉

**Review by alexfauquette — APPROVED** [review]

**Pull request merged**